app/controllers/api/v1/auth_sessions_controller.rb

```diff
@@ -1,41 +1,176 @@
+            respond_with(current_user)
+          else
+            respond_with_error()
+          end
+        else
+          respond_with_error()
+        end
+      end
+
+    else   #..convetional user_id & pwd login....
+      logger.info "... conventonal .user login ..."
+      super
+    end
+  end
+
+
+  private
+
+# -- remove for JWT conversion 
+#  def sign_in_params
+#    params.require(:sign_in).permit :user_id, :password
+#  end
+#
+#  def load_user
+#    @user = User.find_for_database_authentication(user_id: sign_in_params[:user_id])
+#    if @user
+#      return @user
+#    else
+#      render json: {
+#        messages: "Cannot get User",
+#        is_success: false,
+#        data: {}
+#      }, status: :failure
+#    end
+#  end
+
+
+  def respond_with_error
+    render json: {
+      status: { 
+        code: 401, message: 'Invalid credentials.',
+      }
+    }, status: 401
+  end
+
+  def respond_with(current_user, _opts = {})
+    render json: {
+      status: { 
+        code: 200, message: 'Login  successfull.',
+        data: { user: UserSerializer.new(current_user).serializable_hash[:data][:attributes] }
+      }
+    }, status: :ok
+  end
+
+  def respond_to_on_destroy
+    if request.headers['Authorization'].present?
+      jwt_payload = JWT.decode(request.headers['Authorization'].split(' ').last, Rails.application.credentials.devise_jwt_secret_key!).first
+      current_user = User.find(jwt_payload['sub'])
+    end
+    
+    if current_user
       render json: {
-        messages: "Signed In Successfully",
-        is_success: true,
-        data: {user: @user}
+        status: 200,
+        message: 'Logged out successfully.'
       }, status: :ok
     else
       render json: {
-        messages: "Signed In Failed - Unauthorized",
-        is_success: false,
-        data: {}
+        status: 401,
+        message: "Couldn't find an active session."
       }, status: :unauthorized
     end
   end
 
-  private
-  def sign_in_params
-    params.require(:sign_in).permit :user_id, :password
+
+  def decrypt text
+    salt, data = text.split "$$"
+
+    len   = ActiveSupport::MessageEncryptor.key_len
+    key   = ActiveSupport::KeyGenerator.new(
+            Rails.application.secrets.secret_key_base).generate_key salt, len
+    crypt = ActiveSupport::MessageEncryptor.new key
+    crypt.decrypt_and_verify data
   end
 
-  def load_user
-    @user = User.find_for_database_authentication(user_id: sign_in_params[:user_id])
-    if @user
-      return @user
-    else
-      render json: {
-        messages: "Cannot get User",
-        is_success: false,
-        data: {}
-      }, status: :failure
+
+  
+  # De-construct the RRT JWT and check for valid format...
+  # Claims payload consists of:
+  #   iss: https://popps.com"
+  #   sub: ...encrypted user_id and device_guid Hash
+  #   popps_type: "rrt"
+  #
+  # Note: the expiration claim is OPTIONAL - leave it out for perpetual JWT.
```

Comment:
Lines 135 - 141 are a good example of something to consider commenting on the PR and documenting there instead of leaving commented here.